### ISSUE TITLE
"Other" community solutions include the same one I am currently viewing

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/exercism#5652]
![Screenshot 2021-09-02 at 10-07-56 michallepicki #39;s solution for Lasagna in Elixir on Exercism](https://user-images.githubusercontent.com/7852553/131807346-472da7e6-e020-4232-bc24-98d580f1bbdd.png)

**Comment by iHiD** [comment]
Grrrr. There's literally code to stop this. Will work it out. Thanks.

**Comment by NobbZ** [comment]
That is indeed a different solution, than that you are looking at…

The author might be the same, the implementation is different though. The one you are looking at is using magic numbers, the one in "other" module attributes. Perhaps different iterations of the same exercise?

**Comment by angelikatyborska** [comment]
> Perhaps different iterations of the same exercise?

Yes, that's the case. So maybe I am just confused about what's a "solution".

In this UI, I expected to always see the user's latest iteration as their "solution" if they published all of them. That's because people usually settle on "the best attempt" as the last iteration and don't further submit "worse" iterations on purpose. That usually means that earlier iterations are "worse".

**Comment by NobbZ** [comment]
That's why I'm not a friend of "all", "last", "none"…

I've always favored a concept of "attempts" over which I might iterate.

I might start with my first attempt where I just try to solve the exercise to do the next one. I iterate over it until it satisfies the constraints given by the tests, anylizers and perhaps mentors.

Some weeks later I learned a lot more stuff and I start a fresh attempt of this exercise, for which I use clever metaprogramming stuff, not  because it were idiomatic or "better", just because to play with it. I'd like to publish this solution, but perhaps even marked with a big fat "kids, don't try that at home" label.

Currently I also had to publish all the "trash" iterations that lead to the initial publish. Or I had to publish my "kids, don't try at home" solution as the sole one, unpublishing the actual "good" one…
